src/commands/fun/diceroll.ts

```diff
@@ -1,15 +1,36 @@
-import { Message } from 'discord.js';
+import { ApplicationCommandOptionData, ApplicationCommandPermissionData, CommandInteraction, Message } from 'discord.js';
 import { Command } from '@lib/types/Command';
+import { ROLES } from '@root/config';
 
 const DEFAULT_RANGE = [1, 6];
 export default class extends Command {
 
-	description = `Get a random integer within a user-specified range (min and max inclusive). If no range is specified, defaults to a range from ${DEFAULT_RANGE[0]} to ${DEFAULT_RANGE[1]}.`;
+	description = `Get a random integer between \`minimum\` and \`maximum\`.`;
+	extendedHelp = `User specified minimum and maximum are inclusive. If no range is specified, defaults to a range from ${DEFAULT_RANGE[0]} to ${DEFAULT_RANGE[1]}.`;
 	usage = '[min #] | [max #]';
 	aliases = ['random', 'rand', 'dice', 'roll'];
 
-	run(msg: Message, [minimum, maximum]: Array<number>): Promise<Message> {
-		return msg.channel.send(`Your random number is ${Math.floor((Math.random() * (maximum - minimum + 1)) + minimum)}`);
+	options: ApplicationCommandOptionData[] = [
+		{
+			name: 'minimum',
+			description: 'minimum of the roll range',
+			type: 'NUMBER',
+			required: false
+		},
+		{
+			name: 'maximum',
+			description: 'minimum of the roll range',
+			type: 'NUMBER',
+			required: false
+		}
+	]
+
+	run(_msg: Message): Promise<void> { return; }
+
+	tempRun(interaction: CommandInteraction): Promise<void> {
+		const min = interaction.options.getNumber('minimum') || DEFAULT_RANGE[0];
+		const max = interaction.options.getNumber('maximum') || DEFAULT_RANGE[1];
+		return interaction.reply(`Your random number is ${Math.floor((Math.random() * (max - min + 1)) + min)}`);
 	}
 
 	argParser(_msg: Message, input: string): Array<number> {
```

Comment:
This entire command is outdated as I have modified it on my own in the simonCmds branch.